.github/workflows/behat-test.yml

```
@@ -158,12 +158,35 @@ jobs:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
       - name: Download Artifacts
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        uses: actions/download-artifact@f44cd7b40bfd40b6aa1cc1b9b5b7bf03d3c67110 # v4.1.0
         with:
-          name: ${{ inputs.name }}-${{ inputs.os }}-test-reports
+          pattern: ${{ inputs.name }}-${{ inputs.os }}-test-reports-*
           path: ${{ inputs.name }}-xunit-reports
+          merge-multiple: true
 
       - uses: ./.github/actions/publish-report
         with:
           check_name: ${{ inputs.name }}-tests-report
-          path: "${{ inputs.name }}-xunit-reports/**/*.xml"
+          path: '${{ inputs.name }}-xunit-reports/**/*.xml'
+
+  regroup-artifacts:
+    needs: [behat-test-run, behat-test-report]
+    if: always()
```

then Why not moving if here ?
```suggestion
    if: ${{ always() && contains(needs.behat-test-run.result, 'failure') }}
```